aspnetcore/fundamentals/url-rewriting.md

```diff
@@ -0,0 +1,295 @@
+
+![A WebAPI service endpoint has been temporarily changed from version 1 (v1) to version 2 (v2) on the server. A client makes a request to the service at the version 1 path /v1/api. The server sends back a 302 (Found) response with the new, temporary path for the service at version 2 /v2/api. The client makes a second request to the service at the redirect URL. The server responds with a 200 (OK) status code.](url-rewriting/_static/url_redirect.png)
+
+When redirecting requests to a different URL, you will indicate whether the redirect is permanent or temporary. The 301 (Moved Permanently) status code is used where the resource has a new, permanent URL and you wish to instruct the client that all future requests for the resource should use the new URL. The client will cache the response when a 301 status code is received. The 302 (Found) status code is used where the redirection is temporary or generally subject to change, such that the client should not store and reuse the redirect URL in the future. For more information, see [RFC 2616: Status Code Definitions](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html).
+
+A *URL rewrite* is a server-side operation to provide a resource from a different resource address. Rewriting a URL doesn't require a round-trip to the server. The rewritten URL is not returned to the client and won't appear in a browser's address bar. When **http://www.mysite.com/resource** is *rewritten* to **http://www.mysite.com/different-resource**, the client will request **http://www.mysite.com/resource**, and the server will *internally* fetch the resource at **http://www.mysite.com/different-resource**. Although the client might be able to retrieve the resource at the rewritten URL, the client won't be informed that the resource exists at the rewritten URL when it makes its request and receives the response.
+
+![A WebAPI service endpoint has been changed from version 1 (v1) to version 2 (v2) on the server. A client makes a request to the service at the version 1 path /v1/api. The request URL is rewritten to access the service at the version 2 path /v2/api. The service responds to the client with a 200 (OK) status code.](url-rewriting/_static/url_rewrite.png)
+
+## URL Rewriting Middleware
+You can explore the features of the URL Rewriting Middleware with the [URL Rewriting sample application](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/url-rewriting/sample). The application applies rewrite and redirect rules and shows the resultant rewritten or redirected URL.
+
+The examples below rely upon regular expressions (regex) to match and capture the path and segments of the request URL. For more information on using regex, visit the resources in the [Additional Resources](#additional-resources) section and study the examples in the [Regex Examples](#regex-examples) section of this document. 
+
+### When to use URL Rewriting Middleware
```

Comment:
That would break the outline tho ...

* Introduction
  * What is URL Rewriting?
  * Use cases (abstraction: URL from location)
  * URL redirect versus URL rewrite
  * Response Status Codes: 301 (Moved Permanently) vs. 302 (Found)
* URL Rewrite Middleware
  * When to use URL Rewrite Middleware (when not behind IIS or Apache)
  * Package - Microsoft.AspNetCore.Rewrite
  * Extension & Options - `RewriteOptions()` + `app.UseRewriter(options)`
* Additional Resources